src/jdk.internal.vm.ci/share/classes/jdk/vm/ci/hotspot/HotSpotResolvedObjectTypeImpl.java

```diff
@@ -1110,7 +1112,9 @@ public ResolvedJavaField resolveField(UnresolvedJavaField unresolvedJavaField, R
 
     @Override
     public boolean isCloneableWithAllocation() {
-        return (getAccessFlags() & config().jvmAccIsCloneableFast) != 0;
+        HotSpotVMConfig config = config();
+        int miscFlags = UNSAFE.getByte(getKlassPointer() + config.klassMiscFlagsOffset);
+        return (miscFlags & config().jvmAccIsCloneableFast) != 0;
     }
```

Comment:
Maybe introduce getMiscFlags() helper like existing getAccessFlags()?

Reply:
Seems not useful for these two references.